File: backend/app/api/routes/health.py
# GET /api/health - 서버 생존 확인용 엔드포인트.
# 사용법: 로드밸런서/모니터링, 프론트 개발 중 백엔드가 떴는지 확인할 때 curl로 호출.
# 응답 스키마 변경 없이 계속 { "status": "ok" } 형태를 유지하는 게 좋다(외부 헬스체크 계약).

from __future__ import annotations

from fastapi import APIRouter

from app.schemas.health import HealthResponse

router = APIRouter(tags=["health"])


@router.get("/health", response_model=HealthResponse)
async def get_health() -> HealthResponse:
    return HealthResponse(status="ok")

File: frontend/src/api/health.ts
// GET /api/health 호출 래퍼. 현재 UI에서 직접 쓰이진 않지만, 개발 중 백엔드 연결
// 확인용이나 향후 상태 배너 등에 재사용할 수 있게 client.ts 패턴을 따라 만들어둠.

import { apiClient } from "./client";

export function getHealth(): Promise<{ status: string }> {
  return apiClient.get<{ status: string }>("/health");
}
